subsys/dfu/dfu_target/src/dfu_target_stream.c

```diff
@@ -164,7 +156,20 @@ int dfu_target_stream_offset_get(size_t *out)
 
 int dfu_target_stream_write(const uint8_t *buf, size_t len)
 {
+#ifdef CONFIG_DFU_TARGET_STREAM_SAVE_PROGRESS
+	/**
```

Comment:
This (and the alignment requirement) seems to be very tightly coupled with the application protocol assumptions that writes are synchronous and ack-ed data chunked cannot be re-requested. This is however not imposed directly by the `DFU_TARGET_STREAM_SAVE_PROGRESS` feature, so we may need another Kconfig, such as `DFU_TARGET_STREAM_SYNCHRONOUS` to control such a behavior and not force more flexible applications to flush each chunk.

Reply:
Ok, agreed, however for now I've made `CONFIG_DFU_TARGET_STREAM_SYNCHRONOUS`  enabled by default if `CONFIG_DFU_TARGET_STREAM_SAVE_PROGRESS` is enabled.
Looking at the implementation of `stream_flash` it should not increase flash wearing. The biggest issue with flushing immediately is that we do not enable sending chunks which do not have a size aligned to 4 (write block size) but I assume this won't be an issue in most of the cases. So I thought that in most cases we should enable this option, so that there is no strange behavior where the offset after reset is different than the offset before reset.